src/alchemlyb/estimators/ti_.py

```diff
@@ -105,3 +105,45 @@ def fit(self, dHdl):
         self.states_ = means.index.values.tolist()
 
         return self
+
+    def separate_dhdl(self):
+        """
+        For transitions with multiple lambda, this function will separate the
+        dhdl with multiple columns into a list of Dataframe with a single column
+        (single lambda).
+        Returns
+        ----------
+        dHdl_list : list
+            A list of Series such that dHdl_list[k][n] is the potential
+            energy gradient with respect to lambda for each configuration n and
+            lambda k.
+        """
+        if len(self.dhdl.index.names) == 1:
+            # If only one column is present convert to series
+            assert len(self.dhdl.columns) == 1
```

Comment:
Don't use `assert` for error checking. This can be optimized away with `python -O`.

If you want to check for error, use try/except or if and raise the appropriate exception, e.g., `ValueError`.